web/geosearch/datapunt_geosearch/authz.py

```diff
@@ -0,0 +1,135 @@
+import functools
+import json
+import logging
+import re
+import requests
+
+from jwcrypto.jwk import JWKSet
+from jwcrypto.jwt import JWT, JWTMissingKey
+from jwcrypto.common import JWException
+from flask import request as flask_request
+from flask import current_app
+from flask import abort
+from flask import g
+
+logger = logging.getLogger(__name__)
+
+
+def authenticate(func):
+    """
+    Perform authentication check.
+
+    Optionally updates `g.authz_scopes` with scopes available for given token.
+    Returns 401 in case token is invalid.
+    """
+    @functools.wraps(func)
+    def wrapper():
+        check_authentication(request=flask_request)
+        return func()
+    return wrapper
+
+
+def check_authentication(request):
+    """
+    Optionally check authentication via on request.
+
+    Returns None if request contains no Authorization header
+     or header is correct.
+    In case Authorization header is correct - updates `g.authz_scopes` with
+     scopes from claims defined in JWT token.
+    Aborts with 401 in case Authorization header contains incorrect token.
+    """
+    g.authz_scopes = None
```

Comment:
Questions: using `request.authz_scopes` is not a proper Flask practice?

Reply:
In current use case there's no difference, however if we intend to use this code outside of request - using `g` will be much better approach. Therefore use of `g` is future proof solution, and personal preference :)